x-pack/legacy/plugins/index_management/public/app/components/mappings_editor/components/document_fields/fields/edit_field/edit_field.tsx

```diff
@@ -163,6 +155,7 @@ export const EditField = React.memo(({ form, field, allFields, exitEdit }: Props
 
                     {ParametersForm && (
                       <ParametersForm
+                        key={subType ?? type}
```

Comment:
Adding a key gives us the guarantee that we have **a new instance** each time we change the type or subType

Reply:
Can you add this as a comment in the code, along with a note of why we want a new instance?

Reply:
Good idea I will add it. 👍